tests/FSharp.Compiler.ComponentTests/FSharpChecker/TransparentCompiler.fs

```diff
@@ -1061,6 +1061,13 @@ let ``The script load closure should always be evaluated`` useTransparentCompile
     async {
         // The LoadScriptClosure uses the file system shim so we need to reset that.
         let currentFileSystem = FileSystemAutoOpens.FileSystem
+        let assumeDotNetFramework =
+            // The old BackgroundCompiler uses assumeDotNetFramework = true
+            // This is not always correctly loading when this test runs on non-Windows.
+            if not ( System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform( System.Runtime.InteropServices.OSPlatform.Windows)) then
```

Comment:
That doesn't look right. We don't want to assume framework on Windows on .NET 8 for example, probably?

If we want to check if we're running on old CLR or CoreCLR, then it should be something like
```fsharp
System.Runtime.InteropServices.RuntimeInformation.FrameworkDescription.StartsWith(".NET Framework")
```
If we want to check target which it's building/running against (.NET Framework vs SDK), then loaded dll should be checked if it's `mscorlib` or `System.Runtime`.

Reply:
I'm only respecting what is in the codebase today:

https://github.com/dotnet/fsharp/blob/7c20f1904fc5e31b5685b493baf9c9d5e75373f4/src/Compiler/Service/BackgroundCompiler.fs#L1332-L1333

This is something I don't want to deal with in this PR.

Reply:
It is fine to keep this property `assumeDotNetFramework`.

But for this test file, since the ComponentTests are running in a matrix, we should really pass the value according to what Vlad wrote - to exercise the system under test properly (e.g. 1x Windows .Net Framework and 1x Windows NetCore).

Reply:
Updated the test.